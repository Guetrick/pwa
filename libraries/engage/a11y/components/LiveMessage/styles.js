import { css } from 'glamor';

export default css({
  clip: 'rect(1px, 1px, 1px, 1px)',
  height: '1px',
  margin: 0,
  overflow: 'hidden',
  padding: 0,
  position: 'absolute',
  whiteSpace: 'nowrap',
  width: '1px',
  zIndex: -1000,
});
