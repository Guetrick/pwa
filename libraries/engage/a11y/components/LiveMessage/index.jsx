import React from 'react';
import PropTypes from 'prop-types';
import { I18n } from '../../../components';
import styles from './styles';

/**
 * The LiveMessage component can be used to broadcast "aria-live" messages to assistive technology.
 * @param {string} message The message - i18n placeholders are supported.
 * @param {Object} params Optional params for the message.
 * @param {string} 'aria-live' The aria-live type.
 * @returns {JSX}
 */
const LiveMessage = ({ message, params, 'aria-live': ariaLive }) => (
  <div className={styles} role="log" aria-live={ariaLive}>
    <I18n.Text string={message} params={params} />
  </div>
);

LiveMessage.propTypes = {
  message: PropTypes.string.isRequired,
  'aria-live': PropTypes.string,
  params: PropTypes.shape(),
};

LiveMessage.defaultProps = {
  'aria-live': 'polite',
  params: null,
};

export default LiveMessage;
