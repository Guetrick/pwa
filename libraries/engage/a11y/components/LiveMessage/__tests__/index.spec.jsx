import React from 'react';
import { shallow } from 'enzyme';
import LiveMessage from '../index';

jest.mock('../../../../components', () => ({
  I18n: {
    Text: function Translate({ string }) {
      return string;
    },
  },
}));

describe('<LiveMessage />', () => {
  const message = 'My Message';

  it('should render a polite simple message', () => {
    const ariaLive = 'polite';
    const wrapper = shallow(<LiveMessage message={message} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').prop('aria-live')).toBe(ariaLive);
    expect(wrapper.find('Translate').prop('string')).toBe(message);
    expect(wrapper.find('Translate').prop('params')).toEqual(null);
  });

  it('should render an assertive message with params ', () => {
    const ariaLive = 'assertive';
    const params = { my: 'param' };
    const wrapper = shallow(<LiveMessage message={message} params={params} aria-live={ariaLive} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').prop('aria-live')).toBe(ariaLive);
    expect(wrapper.find('Translate').prop('string')).toBe(message);
    expect(wrapper.find('Translate').prop('params')).toEqual(params);
  });
});

