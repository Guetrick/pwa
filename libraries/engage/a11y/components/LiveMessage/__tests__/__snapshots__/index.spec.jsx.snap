// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LiveMessage /> should render a polite simple message 1`] = `
<div
  aria-live="polite"
  className={
    Object {
      "data-css-1krt3b8": "",
    }
  }
  role="log"
>
  <Translate
    params={null}
    string="My Message"
  />
</div>
`;

exports[`<LiveMessage /> should render an assertive message with params  1`] = `
<div
  aria-live="assertive"
  className={
    Object {
      "data-css-1krt3b8": "",
    }
  }
  role="log"
>
  <Translate
    params={
      Object {
        "my": "param",
      }
    }
    string="My Message"
  />
</div>
`;
